testdata/component-naming.txt

```diff
@@ -0,0 +1,42 @@
+# Tests that naming of component fields is correct
+
+openapi-httprequest api.yaml
+cmp api-params.go expect-api-params.go
+
+-- api.yaml --
+openapi: 3.0.0
+info:
+  description: test service description
+  title: test service
+  version: 1.0.0
+
+components:
+  schemas:
+    ErrorCode:
+      type: string
+      description: |-
+        A code representing the kind of error.
+        TODO enumerate all the possible error codes here?
+    Error:
+      type: object
+      properties:
+        code:
+          $ref: "#/components/schemas/ErrorCode"
+
+-- expect-api-params.go --
+// Code generated by openapi-httprequest. DO NOT EDIT.
+
+package params
+
+import (
+	httprequest "gopkg.in/httprequest.v1"
+)
+
+type APIHandler interface {
+}
+
+type Error struct {
+	Code string `json:"code"`
```

Comment:
Ideally this would be:

    Code ErrorCode `json:"code"`

but perhaps that's a different fix?